src/components/implementation/no_interface/new_boot/boot_deps.h

```diff
@@ -0,0 +1,126 @@
+
+/* Assembly function for sinv from new component */
+extern void *__inv_test_entry(int a, int b, int c);
+
+struct cobj_header *hs[MAX_NUM_SPDS+1];
+
+/* The booter uses this to keep track of each comp */
+struct comp_cap_info {
+	struct cos_compinfo *compinfo;
+	vaddr_t addr_start;
+	vaddr_t vaddr_mapped_in_booter;
+	vaddr_t upcall_entry;
+} new_comp_cap_info[MAX_NUM_SPDS+1];
+
+struct cos_compinfo boot_info;
+struct cos_compinfo new_compinfo[MAX_NUM_SPDS+1];
+
+thdcap_t schedule[MAX_NUM_SPDS+1];
+unsigned int sched_cur;
+
+/* Macro for sinv back to booter from new component */
+enum {
+	BOOT_SINV_CAP = round_up_to_pow2(BOOT_CAPTBL_FREE + CAP32B_IDSZ, CAPMAX_ENTRY_SZ)
+};
+
+static vaddr_t
+boot_comp_mmap(int spdid, vaddr_t dest_daddr)
+{
+	vaddr_t addr = cos_page_bump_alloc(&boot_info);
+	assert(addr);
+			
+	if (cos_mem_alias_at(new_comp_cap_info[spdid].compinfo, dest_daddr, &boot_info, addr)) BUG();
+
+	return addr;
+}
+
+static void
+boot_comp_pgtbl_expand(int n_pte, pgtblcap_t pt, vaddr_t vaddr)
+{
+	int j;	
+	for (j = 0 ; j < n_pte ; j++) {
+		if (!cos_pgtbl_intern_alloc(&boot_info, pt, vaddr, SERVICE_SIZE)) BUG();
+	}
+}
+
+
+/* Initialize just the captblcap and pgtblcap, due to hack for upcall_fn addr */
+static void
+boot_compinfo_init(int spdid, captblcap_t *ct, pgtblcap_t *pt, u32_t vaddr)
+{
+	*ct = cos_captbl_alloc(&boot_info);
+	assert(*ct);
+	*pt = cos_pgtbl_alloc(&boot_info);
+	assert(*pt);
+	new_comp_cap_info[spdid].compinfo = &new_compinfo[spdid];
+	cos_compinfo_init(new_comp_cap_info[spdid].compinfo, *pt, *ct, 0, 
+				  (vaddr_t)vaddr, 4, &boot_info);
+}
+
+static void
+boot_newcomp_create(int spdid, captblcap_t ct, pgtblcap_t pt)
+{
+		compcap_t cc;
+		sinvcap_t sinv;
+		thdcap_t main_thd;
+		int i = 0;
+			
+		cc = cos_comp_alloc(&boot_info, ct, pt, (vaddr_t)new_comp_cap_info[spdid].upcall_entry);
+		assert(cc);	
+		new_comp_cap_info[spdid].compinfo->comp_cap = cc;
+	
+		/* Create sinv capability from Userspace to Booter components */
+		sinv = cos_sinv_alloc(&boot_info, boot_info.comp_cap, (vaddr_t)__inv_test_entry);
+		assert(sinv > 0);
+		
+		/* Copy sinv into new comp's capability table at a known location (BOOT_SINV_CAP) */
+		cos_cap_cpy_at(new_comp_cap_info[spdid].compinfo, BOOT_SINV_CAP, &boot_info, sinv);
+
+		main_thd = cos_initthd_alloc(&boot_info, cc);
+		assert(main_thd);
+
+		/* Add created component to "scheduling" array */		
+		while (schedule[i] != NULL) i++;
+		schedule[i] = main_thd;
+}
+
+static void
+boot_bootcomp_init(void)
+{
+	cos_meminfo_init(&boot_info.mi, BOOT_MEM_KM_BASE, COS_MEM_KERN_PA_SZ, BOOT_CAPTBL_SELF_UNTYPED_PT);
+	
+	cos_compinfo_init(&boot_info, BOOT_CAPTBL_SELF_PT, BOOT_CAPTBL_SELF_CT, BOOT_CAPTBL_SELF_COMP, 
+			(vaddr_t)cos_get_heap_ptr(), BOOT_CAPTBL_FREE, &boot_info);
+}
+
+static void
+boot_done(void)
+{
+	printc("Booter: done creating system.\n");
+	cos_thd_switch(schedule[sched_cur]);
+}
+
+void
+boot_thd_done(void)
+{
+	printc("\nWelcome back to the booter!\n");
```

Comment:
Either remove debugging comments, or make their content more easily parse-able by a developer who is not you.